src/providers/SpeechProvider/SpeechProvider.container.js

```diff
@@ -25,29 +29,65 @@ export class SpeechProvider extends Component {
       updateLangSpeechStatus,
       getTtsEngines,
       getTtsDefaultEngine,
+      ttsEngines,
       ttsEngine,
-      setTtsEngine
+      ttsDefaultEngine,
+      setTtsEngine,
+      downloadingLang,
+      setDownloadingLang
     } = this.props;
 
+    const getTtsEngineName = () => {
+      const ttsEnginesNames = ttsEngines?.map(tts => tts.name);
+      if (!ttsEnginesNames) return null;
+      if (
+        downloadingLang?.isdownloading &&
+        downloadingLang.engineName &&
+        ttsEnginesNames.includes(downloadingLang.engineName)
+      )
+        return downloadingLang.engineName;
+
+      if (
+        ttsEngine &&
+        ttsEngine.name &&
+        ttsEnginesNames.includes(ttsEngine.name)
+      )
+        return ttsEngine.name;
+
+      const defaultEngineName = ttsDefaultEngine?.name;
+      if (ttsEnginesNames.includes(defaultEngineName)) return defaultEngineName;
+      return null;
+    };
+
+    let forceChangeVoice = false;
     if (tts.isSupported()) {
       //if android we have to set the tts engine first
-      if (isAndroid()) {
-        getTtsEngines();
-        getTtsDefaultEngine();
+      try {
+        if (isAndroid()) {
+          getTtsEngines();
+          getTtsDefaultEngine();
+        }
+      } catch (error) {
+        console.error(error);
       }
-      if (ttsEngine && ttsEngine.name) {
+
+      const ttsEngineName = getTtsEngineName();
+
+      if (ttsEngineName) {
         try {
-          await setTtsEngine(ttsEngine.name);
+          await setTtsEngine(ttsEngineName);
+          forceChangeVoice = true;
```

Comment:
When the apps starts, the voices always will be updated with the default voice of the tts. The variable `forceChangeVoice` needs to be true only when a new tts was dowloaded.

Reply:
truth! thanks